source/client-backpressure/client-backpressure.md

```diff
@@ -0,0 +1,446 @@
+connection spikes from overloading the system.
+
+#### Ingress Request Rate Limiter
+
+A token bucket based system introduced in MongoDB 8.2 to admit an operation or reject it with a System Overload Error at
+the front door of a mongod/s. It aims to prevent operations spikes from overloading the system.
+
+#### MongoTune
+
+Mongotune is a policy engine outside the server (mongod or mongos) which monitors a set of metrics (MongoDB or system
+host) to dynamically configure MongoDB settings. MongoTune is deployed to Atlas clusters and will dynamically configure
+the connection and request rate limiters to prevent and mitigate overloading the system.
+
+#### RetryableError label
+
+An error is considered retryable if it includes the "RetryableError" label. This error label indicates that an operation
+is safely retryable regardless of the type of operation, its metadata, or any of its arguments.
+
+#### SystemOverloadedError label
+
+An error is considered overloaded if it includes the "SystemOverloadError" label. This error label indicates that the
+server is overloaded. If this error label is present, drivers will backoff before attempting a retry.
+
+#### Overload Errors
+
+An overload error is any command or network error that occurs due to a server overload. For example, when a request
+exceeds the ingress request rate limit:
+
+```js
+{
+  'ok': 0.0,
+  'errmsg': "Rate limiter 'ingressRequestRateLimiter' rate exceeded",
+  'code': 462,
+  'codeName': 'IngressRequestRateLimitExceeded',
+  'errorLabels': ['SystemOverloadedError', 'RetryableError'],
+}
+```
+
+When a new connection attempt exceeds the ingress connection rate limit, the server closes the TCP connection before TLS
+handshake is complete. Drivers will observe this as a network error (e.g. "connection reset by peer" or "connection
+closed").
+
+When a new connection attempt is queued by the server for so long that the driver-side timeout expires, drivers will
+observe this as a network timeout error.
+
+#### Goodput
+
+The throughput of positive, useful output. In the context of drivers, this refers to the number of non-error results
+that the driver processes per unit of time.
+
+See [goodput](https://en.wikipedia.org/wiki/Goodput).
+
+### Requirements for Client Backpressure
+
+#### Overload retry policy
+
+This specification expands the driver's retry ability to all commands, including those not currently considered
+retryable such as updateMany, create collection, getMore, and generic runCommand. The new command execution method obeys
+the following rules:
+
+1. If the command succeeds on the first attempt, drivers MUST deposit `RETRY_TOKEN_RETURN_RATE` tokens.
+    - The value is 0.1 and non-configurable.
+2. If the command succeeds on a retry attempt, drivers MUST deposit `RETRY_TOKEN_RETURN_RATE`+1 tokens.
+3. If a retry attempt fails with an error that does not include `SystemOverloadedError` label, drivers MUST deposit 1
+    token.
```

Comment:
I fail to find item 3 in the pseudocode (lines [109, 155]. Could you please point where it is there?

Reply:
The missing code was added, but incorrectly, as far as I can tell. This thread is superseded by https://github.com/mongodb/specifications/pull/1862#discussion_r2612351104.